packages/core/test-utils/src/fsFixture.js

```diff
@@ -0,0 +1,439 @@
+// @flow strict-local
+
+import type {FileSystem} from '@parcel/fs';
+
+import assert from 'assert';
+import nullthrows from 'nullthrows';
+import path from 'path';
+
+type FixtureToken = {|type: string, value: string|};
+
+type Fixture = FixtureRoot | FixtureChild;
+type FixtureChild = FixtureDir | FixtureFile | FixtureLink;
+
+export function fsFixture(
```

Comment:
Feel like we should move this to its own package? I think this could be useful in projects outside of parcel eventually.